### PR TITLE
Explicity library flushing

### PR DESCRIPTION
**Pull request by simon-id**
## Description

Introduce an opt-in endpoint `GET /flush` that will be called before running test assertions.

## Motivation

5 seconds is too long.

## Workflow

1. ⚠️⚠️ Create your PR as draft
2. Follow the style guidelines of this project (See [how to easily lint the code](https://github.com/DataDog/system-tests/blob/main/docs/edit/lint.md))
3. Work on you PR until the CI passes (if something not related to your task is failing, you can ignore it)
4. Mark it as ready for review

Once your PR is reviewed, you can merge it! :heart:

## Reviewer checklist

* [ ] Check what scenarios are modified. If needed, add the relevant label (`run-parametric-scenario`, `run-profiling-scenario`...). If this PR modifies any system-tests internal, then add the `run-all-scenarios` label ([more info](https://github.com/DataDog/system-tests/blob/main/docs/CI/labels.md)). 
* [ ] CI is green
   * [ ] If not, failing jobs are not related to this change (and you are 100% sure about this statement)
* [ ] if any of `build-some-image` label is present
  1. is the image labl have been updated ? 
  2. just before merging, locally build and push the image to hub.docker.com
* [ ] if a scenario is added (or removed), add (or remove) it in system-test-dasboard nightly

### CODE REVIEW

File: utils/_context/_scenarios.py
Comment (simon-id) [inline utils/_context/_scenarios.py:525]: @cbeauchesne why not do if else here with _wait_interface ?

Comment (simon-id) [reply utils/_context/_scenarios.py:523]: answer: because some tests might want to wait before flushing (time based tests)

File: utils/build/docker/nodejs/express4/app.js
Comment (simon-id) [inline utils/build/docker/nodejs/express4/app.js:257]: TODO: for some reason running this function makes the schema non regression tests fail ???

File: utils/build/docker/nodejs/nextjs/src/app/flush/route.js
Comment (simon-id) [inline utils/build/docker/nodejs/nextjs/src/app/flush/route.js:7]: the nextjs variant is failing because the request to this endpoint is failing. Idk why, it looks like it's throwing an empty error ?

**Comment by Julio-Guerra** [comment]
Libraries don't expose public SDK functions to flush things out. Gracefully stopping the tracer when weblog quits is the most standard approach to me. WDYT of this limitation where libs like dd-trace-go won't be able to implement this "live flushing"?

**Comment by simon-id** [comment]
@Julio-Guerra the nodejs library do not expose a public api to flush stuff either, i had to dig into internals in a very dirty way.
As you said flushing when the process stops is a good idea and it actually is already the case in the Node library! but it's safer to force flushing asynchronously to be sure it's completed, as the ST will wait for the response to that request before running the assertions.
If your language can flush everything reliably at shutdown then go for that sure, but in our case the shutdown flushing is a best effort only.

File: utils/build/docker/nodejs/nextjs/src/app/flush/route.js
Comment (simon-id) [reply utils/build/docker/nodejs/nextjs/src/app/flush/route.js:7]: profiling doesn't work with nexjts apparently

File: docs/internals/flushing.md
Comment (simon-id) [inline docs/internals/flushing.md:16]: need to update that link once the code is merged in master

**Review by cbeauchesne — APPROVED** [review]

File: utils/_context/_scenarios.py
Comment (simon-id) [reply utils/_context/_scenarios.py]: Oh nevermind, this is from an old diff, it's not included in the PR anymore :+1:

**Review by CarlesDD — APPROVED** [review]

**Pull request merged**
